cmd/chain-maind/app/app.go

```diff
@@ -137,7 +137,24 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 		if err != nil {
 			return err
 		}
+
+		tmpFilePath := cleanedPath + ".tmp"
+		tmpFile, err := os.Create(tmpFilePath)
+		if err != nil {
+			return err
+		}
+
 		defer func() {
+			if _, e := os.Stat(tmpFilePath); e == nil {
+				if rmErr := os.Remove(tmpFilePath); rmErr != nil {
+					fmt.Printf("Warning remove tmpFile failed!: %s\n", rmErr)
```

Comment:
removing already opened file can cause some problems.
I recommend after close, and remove it

Reply:
Use WriteFile (github.com/google/renameio) instead, this code has been removed.